tests/UnifiedSpecTests/Util.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace MongoDB\Tests\UnifiedSpecTests;
+
+use MongoDB\Driver\ReadConcern;
+use MongoDB\Driver\ReadPreference;
+use MongoDB\Driver\WriteConcern;
+use stdClass;
+use function array_diff_key;
+use function array_fill_keys;
+use function array_key_exists;
+use function array_keys;
+use function assertContains;
+use function assertEmpty;
+use function assertInternalType;
+use function assertThat;
+use function implode;
+use function isType;
+use function logicalOr;
+
+final class Util
+{
+    public static function assertHasOnlyKeys($arrayOrObject, array $keys)
+    {
+        // TODO: replace isType('object') with instanceOf(stdClass::class)
```

Comment:
My initial reply was going to be: I think `stdClass` is warranted for two reasons:

* In practice, this is only used to validate options after parsing test files.
* `stdClass` is the only object that we can trust to reliably cast as an array for the subsequent `array_diff_key`. Other objects might have private/protected properties.

Then, for I reminded myself that non-public properties still surface in an array cast:

```
>>> (array) new class { private $a; protected $b; public $c; function __construct() { $a=1;$b=2;$c=3; } }
=> [
     "\0class@anonymous\0/home/jmikola/.config/composer/vendor/psy/psysh/src/ExecutionLoopClosure.php(50) : eval()'d code:1$2a4\0a" => null,
     "\0*\0b" => null,
     "c" => null,
   ]
```

In that case, such objects would likely fail if they had those properties, since I don't expect we'd include those keys in the argument. No worries if you'd like to keep this as-is and just delete the TODO.

Reply:
Ah, in that case, let me completely reverse my opinion and suggest we change this to `instanceOf(stdClass::class)` as soon as it's safe to do so